src/bsoncxx/cmake/bsoncxx-config.cmake.in

```diff
@@ -1,3 +1,3 @@
 include(CMakeFindDependencyMacro)
-@BSONCXX_PKG_DEP@
+find_dependency(bson @BSON_REQUIRED_VERSION@ REQUIRED)
```

Comment:
The `REQUIRED` keyword is not necessary for `find_dependency`, IIUC

Reply:
I admit I don't understand how this is meant to work [per docs](https://cmake.org/cmake/help/v3.15/module/CMakeFindDependencyMacro.html):

> `find_dependency` forwards the correct parameters for `QUIET` and `REQUIRED` which were passed to the original `find_package()` call.

I do not see how this call to `find_dependency()` can know what the original parameters to a `find_package()` would have been, so I included `REQUIRED` for assurance. However, local testing appears to enforce the `REQUIRED` behavior even without it being included in this command, so I am fine with removing it.